calibration_details: {image_width_px: 1232, image_height_px: 1072, timestamp: '2025-04-29
    16:32:13', threshold_value: 50, poly_approx_epsilon: 0.01, min_contour_area: 10.0,
  pruning_timestamp: '2025-05-05 12:06:50', pruning_input_file: nozzle_calibration.yaml}
nozzles:
- index: 0
  spray_pattern_pixels:
  - [88.0, 901.0]
  - [160.0, 901.0]
  - [160.0, 984.0]
  - [88.0, 984.0]
- index: 1
  spray_pattern_pixels:
  - [160.0, 900.0]
  - [182.5, 900.0]
  - [182.5, 950.0]
  - [160.0, 950.0]
- index: 2
  spray_pattern_pixels:
  - [182.5, 900.0]
  - [227.25, 900.0]
  - [227.25, 950.0]
  - [182.5, 950.0]
- index: 3
  spray_pattern_pixels:
  - [227.25, 880.0]
  - [282.0, 880.0]
  - [282.0, 951.0]
  - [227.25, 951.0]
- index: 4
  spray_pattern_pixels:
  - [282.0, 875.0]
  - [325.75, 875.0]
  - [325.75, 952.0]
  - [282.0, 952.0]
- index: 5
  spray_pattern_pixels:
  - [325.75, 873.0]
  - [359.75, 873.0]
  - [359.75, 961.0]
  - [325.75, 961.0]
- index: 6
  spray_pattern_pixels:
  - [359.75, 884.0]
  - [407.5, 884.0]
  - [407.5, 958.0]
  - [359.75, 958.0]
- index: 7
  spray_pattern_pixels:
  - [407.5, 871.0]
  - [459.75, 871.0]
  - [459.75, 967.0]
  - [407.5, 967.0]
- index: 8
  spray_pattern_pixels:
  - [459.75, 891.0]
  - [504.5, 891.0]
  - [504.5, 967.0]
  - [459.75, 967.0]
- index: 9
  spray_pattern_pixels:
  - [504.5, 878.0]
  - [546.5, 878.0]
  - [546.5, 959.0]
  - [504.5, 959.0]
- index: 10
  spray_pattern_pixels:
  - [546.5, 862.0]
  - [590.5, 862.0]
  - [590.5, 949.0]
  - [546.5, 949.0]
- index: 11
  spray_pattern_pixels:
  - [590.5, 859.0]
  - [638.25, 859.0]
  - [638.25, 937.0]
  - [590.5, 937.0]
- index: 12
  spray_pattern_pixels:
  - [638.25, 864.0]
  - [682.0, 864.0]
  - [682.0, 935.0]
  - [638.25, 935.0]
- index: 13
  spray_pattern_pixels:
  - [682.0, 893.0]
  - [715.75, 893.0]
  - [715.75, 969.0]
  - [682.0, 969.0]
- index: 14
  spray_pattern_pixels:
  - [715.75, 865.0]
  - [751.25, 865.0]
  - [751.25, 950.0]
  - [715.75, 950.0]
- index: 15
  spray_pattern_pixels:
  - [751.25, 884.0]
  - [802.75, 884.0]
  - [802.75, 959.0]
  - [751.25, 959.0]
- index: 16
  spray_pattern_pixels:
  - [802.75, 870.0]
  - [853.5, 870.0]
  - [853.5, 939.0]
  - [802.75, 939.0]
- index: 17
  spray_pattern_pixels:
  - [853.5, 860.0]
  - [899.0, 860.0]
  - [899.0, 939.0]
  - [853.5, 939.0]
- index: 18
  spray_pattern_pixels:
  - [899.0, 870.0]
  - [939.5, 870.0]
  - [939.5, 945.0]
  - [899.0, 945.0]
- index: 19
  spray_pattern_pixels:
  - [939.5, 873.0]
  - [974.0, 873.0]
  - [974.0, 958.0]
  - [939.5, 958.0]
- index: 20
  spray_pattern_pixels:
  - [974.0, 875.0]
  - [1014.75, 875.0]
  - [1014.75, 947.0]
  - [974.0, 947.0]
- index: 21
  spray_pattern_pixels:
  - [1014.75, 847.0]
  - [1058.25, 847.0]
  - [1058.25, 937.0]
  - [1014.75, 937.0]
- index: 22
  spray_pattern_pixels:
  - [1058.25, 872.0]
  - [1099.25, 872.0]
  - [1099.25, 950.0]
  - [1058.25, 950.0]
- index: 23
  spray_pattern_pixels:
  - [1099.25, 882.0]
  - [1147.25, 882.0]
  - [1147.25, 952.0]
  - [1099.25, 952.0]
- index: 24
  spray_pattern_pixels:
  - [1147.25, 873.0]
  - [1231.0, 873.0]
  - [1231.0, 942.0]
  - [1147.25, 942.0]
